pkgs/top-level/all-packages.nix

```diff
@@ -9628,15 +9628,20 @@ with pkgs;
 
   gtk2 = callPackage ../development/libraries/gtk+/2.x.nix {
     cupsSupport = config.gtk2.cups or stdenv.isLinux;
-    gdktarget = if stdenv.isDarwin then "quartz" else "x11";
     inherit (darwin.apple_sdk.frameworks) AppKit Cocoa;
   };
 
   gtk2-x11 = gtk2.override {
     gdktarget = "x11";
   };
 
-  gtk3 = callPackage ../development/libraries/gtk+/3.x.nix { };
+  gtk3 = callPackage ../development/libraries/gtk+/3.x.nix {
+    inherit (darwin.apple_sdk.frameworks) AppKit Cocoa;
+  };
+
+  gtk3-x11 = gtk3.override {
```

Comment:
Without a use case, I don't think we need to add this here. Packages depending on GTK shouldn't care what backend it has.

Reply:
Well vim fails without x11 and this hurts discoverability, I only knew about this because nix-locate found a header in gtk2-x11. But I removed it for now.